### PR TITLE
refactor(nav-structure): removed __simple-list, added support for -m-horizontal/-m-tertiary at root

### LINKED ISSUE #2973: Refactor nav horizontal
Add `.pf-m-horizontal` as mod to `.pf-m-nav` so that `pf-c-nav__horizontal-list` is `pf-c-nav__list` with updated css variables.

Why we need to do this? 
* As noted here: https://github.com/patternfly/patternfly/pull/2942/files#r409225006
the updates made for scroll button behavior required some extra css set at the root of `pf-c-nav`, which may introduce bugs. If `pf-m-horizontal` is hoisted to `pf-c-nav` root, we can add the new css in the modifier rather than at root.
* It will reduce code significantly if `pf-c-nav__horizontal-list` and `pf-c-nav__tertiary-list` aren't individually styled
* Allow for use of `pf-c-nav__list` or `pf-c-nav__simple-list`/ adds flexibility

### LINKED ISSUE #2996: Nav updates - nav active border colors and others
https://github.com/patternfly/patternfly/pull/2978#issuecomment-617325546

> On the dark version that color is correct, but on the light version we can use the standard active color. We need it that color in dark theme for contrast requirements.

### LINKED ISSUE #3024: Refine nav class structure
Currently the nav HTML on v4 looks like this:
```html
<section class="pf-c-page__main-nav">
  <nav class="pf-c-nav pf-m-horizontal" aria-label="Local">
    <button class="pf-c-nav__scroll-button" disabled aria-label="Scroll left">
      <i class="fas fa-angle-left" aria-hidden="true"></i>
    </button>
    <ul class="pf-c-nav__tertiary-list">
      <li class="pf-c-nav__item">
        <a href="#" class="pf-c-nav__link pf-m-current" aria-current="page">Item 1</a>
      </li>
      <li class="pf-c-nav__item">
        <a href="#" class="pf-c-nav__link">Item 2</a>
      </li>
      <li class="pf-c-nav__item">
        <a href="#" class="pf-c-nav__link">Item 3</a>
      </li>
    </ul>
    <button class="pf-c-nav__scroll-button" disabled aria-label="Scroll right">
      <i class="fas fa-angle-right" aria-hidden="true"></i>
    </button>
  </nav>
</section>
```

We implement something like the tertiary nav in React by having the `<nav>` inspect it's child `<ul class="pf-c-nav__tertiary-list">` to see if it's a horizontal variant and hence should add the class `pf-m-horizontal`.

This would be not ideal but bearable (for now), but the tertiary nav's scroll buttons become wonky outside of a `<section class="pf-c-page__main-nav">`:
image

What if we make the outer `<nav>` element the center-point of our Navigation component and refine the class structure to apply variants like "pf-m-tertiary" for the tertiary nav and "pf-m-light" for use outside a dark background?

### PR DESCRIPTION
**Pull request by mattnolting**
closes #3024 
closes #2996 
closes #2973

This PR 
* replaces `.pf-c-nav__simple-list` with `.pf-c-nav.pf-m-simple`
* adds `.pf-m-light` to `.pf-c-nav.pf-m-horizontal` (we could remove this in favor `.pf-m-tertiary` as `.pf-m-tertiary` can now be used outside of a page section)
* adds a dedicated css var to `.pf-c-nav__link` for `:hover`, `:focus`, `:active` and `.pf-m-current` so that these variables can be overwritten without wrapping in `.pf-c-nav__item:state {}`

## Breaking changes
summary

The following classes have been removed. Any reference to them should be removed as they are no longer used in patternfly:
* `.pf-c-`

The following variables have been removed. Any reference to them should be removed as they are no longer used in patternfly:
* `--`

The following variables have been renamed. Any reference to the old variable should be updated to use the new variable:
* `--pf-c-` has changed to `--pf-c-`

Some of the general CSS structure has also changed, so please reference the [file changeset](https://github.com/patternfly/patternfly/pull/-/files).

### CODE REVIEW

**Comment by redallen** [comment]
So https://github.com/patternfly/patternfly/pull/3055 should be closed?

**Comment by mattnolting** [comment]
@redallen if we choose to accept these changes, yes [3055](https://github.com/patternfly/patternfly/pull/3055) should be closed. But I wanted to keep it open until we have a chance to discuss both approaches.

File: src/patternfly/components/Nav/examples/Nav.md [hunk 1]
```diff
@@ -515,23 +491,23 @@ The navigation system relies on several different sub-components:
...
-| `.pf-c-nav__simple-list` | `<ul>` | Initiates simple nav list. |
-| `.pf-c-nav__item` | `<li>` | Initiates default nav list item. |
-| `.pf-c-nav__scroll-button` | `<button>` | Intitiates a nav scroll button. **Required for horizontal navs** |
-| `.pf-c-nav__link` | `<a>` | Initiates default nav list link. |
+| `.pf-c-nav__list` | `<ul>` | Initiates nav list. |
+| `.pf-c-nav__item` | `<li>` | Initiates nav list item. |
+| `.pf-c-nav__link` | `<a>` | Initiates nav list link. |
 | `.pf-c-nav__section` | `<section>` | Initiates a nav section element. |
 | `.pf-c-nav__section-title` | `<h1>`, `<h2>`, `<h3>`, `<h4>`, `<h5>`, `<h6>` | Initiates a nav section title. |
 | `.pf-c-nav__toggle` | `<span>` | Initiates a chevron indicating expandability of a `pf-c-nav__list-link`. |
```
Comment (christiemolloy) [inline src/patternfly/components/Nav/examples/Nav.md:504]:
```suggestion
| `.pf-c-nav__toggle` | `<span>` | Initiates the nav toggle icon wrapper. |
```

File: src/patternfly/components/Nav/nav.scss [hunk 2]
```diff
@@ -525,28 +563,6 @@
...
-
-    &.pf-m-current {
-      --pf-c-nav__link--after--BorderLeftWidth: var(--pf-c-nav__subnav__link--m-current--after--BorderWidth);
-      --pf-c-nav__link--after--BorderColor: var(--pf-c-nav__subnav__link--m-current--after--BorderColor);
-    }
-  }
-
   &::-webkit-scrollbar {
     display: none;
     visibility: hidden;
```
Comment (mcoker) [inline src/patternfly/components/Nav/nav.scss:568]: Should we use the `@pf-overflow-hide-scroll` mixin here?

File: src/patternfly/components/Nav/examples/Nav.md [hunk 3]
```diff
@@ -515,23 +491,23 @@ The navigation system relies on several different sub-components:
...
 | `.pf-c-nav__toggle-icon` | `<span>` | Initiates a nav toggle icon wrapper. |
+| `.pf-c-nav__scroll-button` | `<button>` | Initiates a nav scroll button. **Required for horizontal navs** |
+| `.pf-m-horizontal` | `.pf-c-nav` | Modifies nav for the horizontal variation. |
+| `.pf-m-tertiary` | `.pf-c-nav` | Modifies nav for the tertiary variation. |
+| `.pf-m-light` | `.pf-c-nav` | Modifies nav for the light variation. **Note: for use with vertical nav, `.pf-m-light` is required on the page component's sidebar element (`.pf-c-page__sidebar`)**. |
```
Comment (mcoker) [inline src/patternfly/components/Nav/examples/Nav.md:509]: Should we change this back to what it was since it doesn't apply to horizontal nav?

Comment (mattnolting) [reply src/patternfly/components/Nav/examples/Nav.md]: Yes!

**Comment by mattnolting** [comment]
@mcoker @christiemolloy updated!

**Pull request merged**